flyteplugins/go/tasks/plugins/webapi/agent/plugin.go

```diff
@@ -402,31 +402,35 @@ func buildTaskExecutionMetadata(taskExecutionMetadata core.TaskExecutionMetadata
 	}
 }
 
-func newAgentPlugin(agentService *core.AgentService) webapi.PluginEntry {
+func newConnectorPlugin(connectorService *core.AgentService) webapi.PluginEntry {
 	ctx := context.Background()
-	cfg := GetConfig()
-	clientSet := getAgentClientSets(ctx)
-	agentRegistry := getAgentRegistry(ctx, clientSet)
-	supportedTaskTypes := maps.Keys(agentRegistry)
+	gob.Register(ResourceMetaWrapper{})
+	gob.Register(ResourceWrapper{})
+
+	clientSet := getConnectorClientSets(ctx)
+	connectorRegistry := getConnectorRegistry(ctx, clientSet)
+	supportedTaskTypes := maps.Keys(connectorRegistry)
+	connectorService.SetSupportedTaskType(supportedTaskTypes)
+
+	plugin := &Plugin{
+		metricScope: promutils.NewScope("connector_plugin"),
+		cfg:         GetConfig(),
+		cs:          clientSet,
+		registry:    connectorRegistry,
+	}
+	plugin.watchConnectors(ctx, connectorService)
 
 	return webapi.PluginEntry{
-		ID:                 "agent-service",
+		ID:                 "connector-service",
```

Comment:
I think we still need to register two plugins (agent-service & connector-service) here, so the agent service should still work when people upgrade propeller